[RubyNguyen07] iP

RubyNguyen07 opened the pull request:
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

### DukePro
DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn 
- ~super~supersuper-fast to use

All you need to do is: 
1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html) 
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :yum:

Features:
- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

RubyNguyen07 committed:
Change Duke file so that Duke can repeat and exit when needed

RubyNguyen07 committed:
Duke can store items and return them as list later

RubyNguyen07 committed:
Duke can mark item as done or undone

RubyNguyen07 committed:
Duke can track 3 types of tasks

RubyNguyen07 committed:
Add Automated Text UI Testing

RubyNguyen07 committed:
Duke can handle a few exceptions

RubyNguyen07 committed:
There is now tests for Duke's ability to handle exceptions

RubyNguyen07 committed:
Duke can delete  tasks

RubyNguyen07 committed:
Duke can now be tested with its new delete feature

RubyNguyen07 committed:
There are JavaDoc for most public methods

RubyNguyen07 committed:
Fix deadline bug and add UI to separate commands

RubyNguyen07 committed:
Save

RubyNguyen07 committed:
Dates and Times

RubyNguyen07 committed:
Merge dates and times feature
Merge branch 'branch-Level-8'

RubyNguyen07 committed:
OOP implemented

RubyNguyen07 committed:
Organize into java packages

RubyNguyen07 committed:
Merge branch 'add-gradle-support'

RubyNguyen07 committed:
gradle added

RubyNguyen07 committed:
Add JUnit tests

RubyNguyen07 committed:
Create a Jar file

RubyNguyen07 committed:
Level-9

RubyNguyen07 committed:
Coding standard

RubyNguyen07 committed:
Change coding standard and resolve conflicts

RubyNguyen07 committed:
Add Checkstyle to control coding style

RubyNguyen07 committed:
Add UI using JavaFX

RubyNguyen07 committed:
Change UI to look nicer

CarrieLi1015 commented on src/main/java/duke/Parser.java:
```diff
@@ -1,7 +1,8 @@
 
-
+//CHECKSTYLE:OFF
 import duke.exceptions.*;
```
Here might be better to follow the Java coding standard. The usage of wildcard import is not recommended.

CarrieLi1015 commented on src/main/java/duke/Ui.java:
```diff
@@ -37,40 +38,45 @@ public void close() throws IOException {
 
-    public void listCommand() {
+    public String listCommand() {
         ArrayList<Task> arr = TaskList.getList();
```
When using ArrayList, it might be better to use plural form for the variable name. Besides, arr might not be a very meaning variable name, might consider change it.

CarrieLi1015 commented on src/main/java/duke/Ui.java:
```diff
@@ -106,86 +113,95 @@ static void deadlineCommand(String str, Storage storage) throws IOException {
-     void unmarkCommand(String str, Storage storage) throws IOException {
+     String unmarkCommand(String str, Storage storage) throws IOException {
         try {
             int index = Parser.getTaskNumber(str,7) - 1;
```
Might be better to have a space between "," and "7".

CarrieLi1015 commented on src/main/java/duke/Ui.java:
```diff
@@ -1,5 +1,6 @@
 package duke;
 
+
 import duke.exceptions.*;
```
Same wildcard issue mentioned above.

CarrieLi1015 reviewed (commented):
Overall well done! Just some minor details need to take note. 👍

sachindavid96 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,21 @@
+public class Task {
+    String taskName;
+    boolean isDone;
```
Perhaps the boolean isDone can be initialized to false, since when a task is first created, it is usually not completed

sachindavid96 commented on src/main/java/Duke.java:
```diff
@@ -12,27 +14,57 @@ public static void main(String[] args) throws IOException {
                 int cnt = 1;
-                while (cnt <= curr) {
-                    System.out.println(cnt + ". " + arr[cnt - 1]);
+                while (cnt <= arr.size()) {
```
The variable 'cnt' can be named in a more descriptive manner

sachindavid96 commented on src/main/java/Duke.java:
```diff
@@ -12,27 +14,57 @@ public static void main(String[] args) throws IOException {
+                    System.out.println(cnt + ". " + "[" +  item.getStatus() + "] " + item.taskName);
                     cnt++;
                 }
+            } else if (str.length() >= 4 && str.substring(0, 4).equals("mark")) {
```
It would be more elegant if this if-else block were to be replaced by a switch statement

RubyNguyen07 committed:
Add JavaDoc

Add the JavaDoc for the function splitString in Storage
so that its purpose is clear

RubyNguyen07 committed:
Add assertions

Add assertions at various points to check if code behaves as it should be

RubyNguyen07 committed:
Merge pull request #1 from RubyNguyen07/branch-A-Assertions

Add assertions

RubyNguyen07 committed:
Check code quality

Enforce Happy path in the code

RubyNguyen07 committed:
Merge pull request #2 from RubyNguyen07/branch-A-CodeQuality

Check code quality

RubyNguyen07 committed:
Add Note feature

Add Note feature so that users can note trivial things down

RubyNguyen07 committed:
Merge pull request #3 from RubyNguyen07/branch-D-Extension

Add Note feature

RubyNguyen07 committed:
Add User Guide

RubyNguyen07 committed:
Change Readme

RubyNguyen07 committed:
Fix readme file

RubyNguyen07 committed:
Fix readme file
